scrapy/settings/__init__.py

```diff
@@ -83,7 +83,8 @@ class BaseSettings(MutableMapping):
     def __init__(self, values=None, priority='project'):
         self.frozen = False
         self.attributes = {}
-        self.update(values, priority)
+        if values is not None:
```

Comment:
Nice catch.

Since we are at it, maybe it would make sense to just use `if values:`. I imagine empty settings do not require `self.update` either.

Reply:
> Nice catch.
> 
> Since we are at it, maybe it would make sense to just use `if values:`. I imagine empty settings do not require `self.update` either.

Agree.